### PR TITLE
Added vnet links to central private dns zones

### PR DESCRIPTION
**Pull request by chrisorisawayi**
### Change description ###

To support private endpoint DNS in use by the MI Data Platform, vnet links for the below private dns zones added for all environments:

-   privatelink.database.windows.net
-   privatelink.blob.core.windows.net
-   privatelink.vaultcore.azure.net
-   privatelink.datafactory.azure.net
-   privatelink.postgres.database.azure.com

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: config/terraform/ss/vars/01-network/demo.tfvars [hunk 1]
```diff
@@ -7,7 +7,13 @@ iaas_subnet_cidr_blocks                = "10.51.96.0/25"
 application_gateway_subnet_cidr_blocks = "10.51.96.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["demo.platform.hmcts.net"]
+private_dns_zones = [
+  "demo.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/demo.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/dev.tfvars [hunk 2]
```diff
@@ -7,7 +7,13 @@ iaas_subnet_cidr_blocks                = "10.145.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.145.32.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["dev.platform.hmcts.net"]
+private_dns_zones = [
+  "dev.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/dev.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/ithc.tfvars [hunk 3]
```diff
@@ -7,6 +7,12 @@ iaas_subnet_cidr_blocks                = "10.143.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.143.32.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["ithc.platform.hmcts.net"]
+private_dns_zones = [
+  "ithc.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/ithc.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/prod.tfvars [hunk 4]
```diff
@@ -7,6 +7,12 @@ iaas_subnet_cidr_blocks                = "10.144.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.144.32.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["platform.hmcts.net"]
+private_dns_zones = [
+  "platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/prod.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/sbox.tfvars [hunk 5]
```diff
@@ -7,6 +7,12 @@ iaas_subnet_cidr_blocks                = "10.140.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.140.32.128/25"
 
 private_dns_subscription = "1497c3d7-ab6d-4bb7-8a10-b51d03189ee3"
-private_dns_zones        = ["sandbox.platform.hmcts.net"]
+private_dns_zones = [
+  "sandbox.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/sbox.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/stg.tfvars [hunk 6]
```diff
@@ -7,6 +7,12 @@ iaas_subnet_cidr_blocks                = "10.142.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.142.32.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["staging.platform.hmcts.net"]
+private_dns_zones = [
+  "staging.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/stg.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

File: config/terraform/ss/vars/01-network/test.tfvars [hunk 7]
```diff
@@ -7,6 +7,12 @@ iaas_subnet_cidr_blocks                = "10.141.32.0/25"
 application_gateway_subnet_cidr_blocks = "10.141.32.128/25"
 
 private_dns_subscription = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-private_dns_zones        = ["test.platform.hmcts.net"]
+private_dns_zones = [
+  "test.platform.hmcts.net",
+  "privatelink.database.windows.net",
+  "privatelink.blob.core.windows.net",
+  "privatelink.vaultcore.azure.net",
```
Comment (timja) [inline config/terraform/ss/vars/01-network/test.tfvars:14]:
```suggestion
  "privatelink.vaultcore.azure.net",
  "privatelink.postgres.database.azure.com",
```

**Review by timja — APPROVED** [review]
Could you also add `privatelink.postgres.database.azure.com` while you're here? we need to do it soon but since you're going to need to roll out all to all envs makes sense to do this now

File: config/terraform/ss/vars/01-network/ithc.tfvars (hunk 3 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/ithc.tfvars:14]: postgres added

File: config/terraform/ss/vars/01-network/demo.tfvars (hunk 1 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/demo.tfvars:14]: postgres link added

File: config/terraform/ss/vars/01-network/prod.tfvars (hunk 4 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/prod.tfvars:14]: postgres link added

File: config/terraform/ss/vars/01-network/sbox.tfvars (hunk 5 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/sbox.tfvars:14]: postgres link added

File: config/terraform/ss/vars/01-network/stg.tfvars (hunk 6 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/stg.tfvars:14]: postgres link added

File: config/terraform/ss/vars/01-network/test.tfvars (hunk 7 above)
Comment (chrisorisawayi) [reply config/terraform/ss/vars/01-network/test.tfvars:14]: postgres link added

**Pull request merged**